envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/listeners/filters/RBACFilterFactory.kt

```diff
@@ -130,11 +132,28 @@ class RBACFilterFactory(
     }
 
     private fun mapClientToPrincipals(client: String, snapshot: GlobalSnapshot): List<Principal> {
-        if (client !in incomingPermissionsProperties.sourceIpAuthentication.enabledForServices) {
-            return headerPrincipals(client)
+        val staticRangesForClient = staticIpRanges[client]
+
+        return if (client in incomingPermissionsProperties.sourceIpAuthentication.enabledForServices) {
+            sourceIpPrincipals(client, snapshot)
+        } else if (!staticRangesForClient.isNullOrEmpty()) {
```

Comment:
Maybe it would be better to change the condition to `staticRangesForClient != null`? If the service name is defined in staticIpRanges with empty list, I would rather expect, that static ip authentication is enabled, but no IP will be accepted. But maybe it is only me.